Federation library failing to compile with tsc

<!--

Thanks for taking the time to file a helpful issue report!

Before reporting a bug, please first check for existing or closed issues
and read the instructions for filing a bug report:

https://github.com/apollographql/federation/blob/HEAD/CONTRIBUTING.md#reporting-bugs

### This bug report should include:

- [x] A short, but descriptive title. The title doesn't need "Apollo" in it.
- [x] When a published package is being used, the package management system
      it is published in along with the specific names and versions.
- [x] If applicable, the last version(s) where the problem did _not_ occur.
- [x] The expected behavior.
- [x] The actual behavior.
- [ ] A **simple**, runnable reproduction!
- [ ] The federation version you are targeting.
      Please make a GitHub repository that anyone can clone and run and see the
      problem.

Please make sure that you include the following information to ensure that
your issue is actionable. If you don't follow the above template, your issue
may end up being closed without anyone looking at it carefully.

We appreciate your help!

-->

## Last working version
@apollo/federation@0.33.9

## Broken versions
Anything newer than @apollo/federation@0.33.9

## Expected Behaviour
Apollo federation should compile with `tsc -p tsconfig.json --noEmit`

## Actual Behaviour
Apollo federation fails to compile due to composition/utils.d.ts referencing a namespace that the compiler cannot find.
```
node_modules/@apollo/federation/dist/composition/utils.d.ts:43:371 - error TS2503: Cannot find namespace 'Kind'.

43     kind: (Kind.SCALAR_TYPE_DEFINITION | Kind.OBJECT_TYPE_DEFINITION | Kind.INTERFACE_TYPE_DEFINITION | Kind.UNION_TYPE_DEFINITION | Kind.ENUM_TYPE_DEFINITION | Kind.INPUT_OBJECT_TYPE_DEFINITION | Kind.DIRECTIVE_DEFINITION | Kind.SCALAR_TYPE_EXTENSION | Kind.OBJECT_TYPE_EXTENSION | Kind.INTERFACE_TYPE_EXTENSION | Kind.UNION_TYPE_EXTENSION | Kind.ENUM_TYPE_EXTENSION | Kind.INPUT_OBJECT_TYPE_EXTENSION)[];
```

I don't have the time right now to put together a reproduction repo, but if this is required I can look into doing it at a later point. I've seen people mentioning this same issue on stack overflow but can't see an issue anywhere in the issue tracker.



## Typescript Config
I've added the typescript config that we use as that's the main thing I can think of that might influence this to fail.
```
{
  "compilerOptions": {
    "target": "es2017",
    "module": "commonjs",
    "strict": true,
    "baseUrl": ".",
    "paths": {
      "*": ["src/*"],
    },
    "typeRoots": ["node_modules/@types"],
    "types": ["node", "jest"],
    "esModuleInterop": true,
    "inlineSourceMap": true,
    "allowJs": false,
    "resolveJsonModule": true,
    "strictFunctionTypes": false,
  }
}
```

--- a/federation-js/CHANGELOG.md
+++ b/federation-js/CHANGELOG.md
@@ -6,7 +6,7 @@ This CHANGELOG pertains only to Apollo Federation packages in the `0.x` range. T
 
 > The changes noted within this `vNEXT` section have not been released yet.  New PRs and commits which introduce changes should include an entry in this `vNEXT` section as part of their development.  When a release is being prepared, a new header will be (manually) created below and the appropriate changes within that release will be moved into the new section.
 
-- Fixes typing incompatibility issue with graphql15. [PR #1585](https://github.com/apollographql/federation/pull/1585)
+- Fixes type incompatibility issue with graphql15. [PR #1585](https://github.com/apollographql/federation/pull/1585)
 
 ## v0.35.3
 

--- a/federation-js/CHANGELOG.md
+++ b/federation-js/CHANGELOG.md
@@ -6,7 +6,7 @@ This CHANGELOG pertains only to Apollo Federation packages in the `0.x` range. T
 
 > The changes noted within this `vNEXT` section have not been released yet.  New PRs and commits which introduce changes should include an entry in this `vNEXT` section as part of their development.  When a release is being prepared, a new header will be (manually) created below and the appropriate changes within that release will be moved into the new section.
 
-- _Nothing yet! Stay tuned._
+- Fixes typing incompatibility issue with graphql15. [PR #1585](https://github.com/apollographql/federation/pull/1585)
 
 ## v0.35.3
 

--- a/federation-js/src/composition/utils.ts
+++ b/federation-js/src/composition/utils.ts
@@ -638,7 +638,7 @@ export function diffTypeNodes(
       ? []
       : [firstNode.name.value, secondNode.name.value];
 
-  const kindDiff =
+  const kindDiff: Kind[keyof Kind][] =
     firstNode.kind === secondNode.kind ? [] : [firstNode.kind, secondNode.kind];
 
   return {

--- a/federation-js/src/composition/utils.ts
+++ b/federation-js/src/composition/utils.ts
@@ -638,7 +638,7 @@ export function diffTypeNodes(
       ? []
       : [firstNode.name.value, secondNode.name.value];
 
-  const kindDiff: Kind[keyof Kind][] =
+  const kindDiff: typeof Kind[keyof typeof Kind][] =
     firstNode.kind === secondNode.kind ? [] : [firstNode.kind, secondNode.kind];
 
   return {

